dataprofiler/tests/data_readers/test_json_data.py

```diff
@@ -45,6 +45,21 @@ def setUpClass(cls):
             dict(path=os.path.join(test_dir, "json/honeypot_intentially_mislabeled_file.csv"), encoding='utf-8', count=14),
             dict(path=os.path.join(test_dir, "json/honeypot_intentially_mislabeled_file.parquet"), encoding='utf-8', count=14)
         ]
+
+        cls.buffer_list = []
+        for input_file in cls.input_file_names:
+            # add BytesIO
+            buffer_info = input_file.copy()
+            with open(input_file['path'], 'rb') as fp:
+                buffer_info['path'] = BytesIO(fp.read())
+            cls.buffer_list.append(buffer_info)
+
+        cls.file_or_buf_list = cls.input_file_names + cls.buffer_list
+
+    @classmethod
+    def setUp(self):
+        for buffer in self.buffer_list:
+            buffer['path'].seek(0)
     
     def test_is_match_for_string_streams(self):
```

Comment:
can we make a generic `is_match` that runs through `self.file_or_buf_list`? Then we can get rid of the duplicate functions as well.
can just call it `test_is_match`